File: public/docs/_examples/server-communication/ts/app/wiki/wiki.component.ts
// #docregion
import { Component }        from '@angular/core';
import { Observable }       from 'rxjs/Observable';

import { WikipediaService } from './wikipedia.service';

@Component({
  moduleId: module.id,
  selector: 'my-wiki',
  templateUrl: 'wiki.component.html',
  providers: [ WikipediaService ]
})
export class WikiComponent {
  title   = 'Wikipedia Demo';
  fetches = 'Fetches after each keystroke';
  items: Observable<string[]>;

  search (term: string) {
    this.items = this.wikipediaService.search(term);
  }

  constructor (private wikipediaService: WikipediaService) { }
}

File: public/docs/_examples/server-communication/dart/lib/wiki/wiki_component.dart
// #docregion
import 'dart:async';

import 'package:angular2/core.dart';

import 'wikipedia_service.dart';

@Component(
    selector: 'my-wiki',
    template: '''
      <h1>Wikipedia Demo</h1>
      <p><i>Fetches after each keystroke</i></p>
      <input #term (keyup)="search(term.value)"/>
      <ul>
        <li *ngFor="let item of items">{{item}}</li>
      </ul>
    ''',
    providers: const [WikipediaService])
class WikiComponent {
  final WikipediaService _wikipediaService;
  List items = [];

  WikiComponent(this._wikipediaService);

  Future search(String term) async {
    items = await this._wikipediaService.search(term);
  }
}
